src/viewport.cpp

```diff
@@ -1036,6 +1038,33 @@ static TileHighlightType GetTileHighlightType(TileIndex t)
 		}
 	}
 
+	if (_current_company < MAX_COMPANIES && _infrastructure_window_open) {
+		/* Edge case of company owned tramway on non-company owned tile */
+		if (IsTileType(t, MP_ROAD) && !IsRoadDepot(t) && HasTileRoadType(t, RTT_TRAM)) {
+			if (IsRoadOwner(t, RTT_TRAM, _current_company)) {
+				return THT_WHITE;
+			}
+		}
+
+		switch (GetTileType(t)) {
+			case MP_ROAD:
+			case MP_RAILWAY:
+			case MP_TUNNELBRIDGE:
```

Comment:
Level-crossings can also have multiple owners, one for the rail, one for the road and one for tram...

I'm not sure about bridges, they probably also have road and tram owners.